brave/src/main/java/brave/propagation/ThreadLocalCurrentTraceContext.java

```diff
@@ -92,12 +92,21 @@ public void clear() {
   @Override public Scope newScope(@Nullable TraceContext currentSpan) {
     final TraceContext previous = local.get();
     local.set(currentSpan);
-    class ThreadLocalScope implements Scope {
-      @Override public void close() {
-        local.set(previous);
-      }
-    }
-    Scope result = new ThreadLocalScope();
+    Scope result = new ThreadLocalScope(local, previous);
     return decorateScope(currentSpan, result);
   }
+
+  static final class ThreadLocalScope implements Scope {
```

Comment:
This seems nice - seems like it could fix something?

Reply:
well  I'll update.. with where I was going.. basically revert to null can be a singleton. I still can't get the thing to unload when leaked as the cover type is brave's.

Reply:
ptal